libvcx/src/messages/get_message.rs

```diff
@@ -539,55 +540,56 @@ mod tests {
-
-        let _all_messages = download_messages(None, None, None).unwrap();
-
-        let pending = download_messages(None, Some(vec!["MS-103".to_string()]), None).unwrap();
-        assert_eq!(pending.len(), 1);
-        assert!(pending[0].msgs[0].decrypted_payload.is_some());
-
-        let accepted = download_messages(None, Some(vec!["MS-104".to_string()]), None).unwrap();
-        assert_eq!(accepted[0].msgs.len(), 2);
-
-        let specific = download_messages(None, None, Some(vec![accepted[0].msgs[0].uid.clone()])).unwrap();
+        send_generic_message(alice_to_faber, "Hello Faber", &json!({"msg_type": "tofaber", "msg_title": "msg1"}).to_string()).unwrap();
+
+        info!("test_download_messages :: going to download all messages");
+        let all_messages = download_messages(None, None, None).unwrap();
+        info!("_all_messages = {}", serde_json::to_string(&all_messages).unwrap());
+        assert_eq!(all_messages.len(), 1);
+        assert_eq!(all_messages[0].msgs.len(), 3);
+        assert!(all_messages[0].msgs[0].decrypted_payload.is_some());
+        assert!(all_messages[0].msgs[1].decrypted_payload.is_some());
+
+        let received = download_messages(None, Some(vec![MessageStatusCode::Received.to_string()]), None).unwrap();
+        assert_eq!(received.len(), 1);
+        assert_eq!(received[0].msgs.len(), 2);
+        assert!(received[0].msgs[0].decrypted_payload.is_some());
+        assert_eq!(received[0].msgs[0].status_code, MessageStatusCode::Received);
+        assert!(received[0].msgs[1].decrypted_payload.is_some());
+
+        // there should be review aries message connections/1.0/response from Aries-Faber connection protocol
+        let reviewed = download_messages(None, Some(vec![MessageStatusCode::Reviewed.to_string()]), None).unwrap();
+        assert_eq!(reviewed.len(), 1);
+        assert_eq!(reviewed[0].msgs.len(), 1);
+        assert!(reviewed[0].msgs[0].decrypted_payload.is_some());
+        assert_eq!(reviewed[0].msgs[0].status_code, MessageStatusCode::Reviewed);
+
+        let rejected = download_messages(None, Some(vec![MessageStatusCode::Rejected.to_string()]), None).unwrap();
+        assert_eq!(rejected.len(), 1);
+        assert_eq!(rejected[0].msgs.len(), 0);
+
+        let specific = download_messages(None, None, Some(vec![received[0].msgs[0].uid.clone()])).unwrap();
         assert_eq!(specific.len(), 1);
+        assert_eq!(specific[0].msgs.len(), 1);
 
-        // No pending will return empty list
-        let empty = download_messages(None, Some(vec!["MS-103".to_string()]), Some(vec![accepted[0].msgs[0].uid.clone()])).unwrap();
-        assert_eq!(empty.len(), 1);
-
-        let hello_msg = download_messages(None, None, Some(vec![hello_uid])).unwrap();
-        assert_eq!(hello_msg[0].msgs[0].decrypted_payload, Some("{\"@type\":{\"name\":\"hello\",\"ver\":\"1.0\",\"fmt\":\"json\"},\"@msg\":\"hello\"}".to_string()));
-
-        // Agency returns a bad request response for invalid dids
```

Comment:
Does this require changes on agency side?

Reply:
Node agency is currently not performing did validation, it just tries to match the did. We should probably audit agency inputs and probably think about its error reporting in general...